usr/sbin/p11sak/p11sak.c

```diff
@@ -1656,6 +1657,12 @@ static CK_RV parse_list_key_args(char *argv[], int argc, p11sak_kt *kt,
         } else if (strcmp(argv[i], "--slot") == 0) {
             if (i + 1 < argc) {
                 *slot = (CK_ULONG) atol(argv[i + 1]);
+                if ((*slot > 0) || (strcmp(argv[i + 1], "0") == 0)) {
```

Comment:
just a minor thing: If the slot is specified as '+0' or '0x0', or ' 0', then you would still treat this as an error. However, this might be acceptable.....

Reply:
atol returns 0 if the string cannot be converted to a long int. It returns 0, too, if the argument is "0". Hence, I check for the string to be "0" or for the long to be greater than 0. The user input " 0" sets the slotID = 0. Others I have not tested.

Reply:
right, atoi also accepts hex format (e.g. 0x0), octal, prefixed with whitespace chars, etc. So it might return 0 also for a valid zero input test, but your strcmp with "0" might not catch that.  You might want to use strtol() instead or atoi(), to better detect parsing errors:
From https://man7.org/linux/man-pages/man3/strtol.3.html
```
           errno = 0;    /* To distinguish success/failure after call */
           val = strtol(str, &endptr, base);

           /* Check for various possible errors */

           if (errno != 0) {
               perror("strtol");
               exit(EXIT_FAILURE);
           }

           if (endptr == str) {
               fprintf(stderr, "No digits were found\n");
               exit(EXIT_FAILURE);
           }

           /* If we got here, strtol() successfully parsed a number */
```

Reply:
I changed the parsing of the slot number replacing atol() with strtol().